[kervyntan] iP

kervyntan opened the pull request:
# Kervyn

> The way to get started is to quit talking and begin doing - Walt Disney

## A revolutionary chatbot, designed to take the load of task tracking off you, making your life 10x more organized and carefree.
- 3 Types of events: Todos, Deadlines & Events
- List out your tasks 
- Find the task you need with a simple keyword search!

**Dying** to get started? Well..
1. Download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. Double-click it.
3. Add your tasks based on the different types available (todo, deadline or event!)
4. Play around with the different commands and be blown away :exploding_head:

Features:
- [x] Add your tasks!
- [x] List and see the tasks you currently have stored!
- [x] Mark/unmark your completed/uncompleted tasks for reference in the future
- [ ] Friendly GUI (coming soon) to interact with
... and many more!  

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
    public static void main(String[] args) throws IOException {
        new Kervyn("data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

kervyntan committed:
Added content for Level-0

kervyntan committed:
Added content for Level-1

kervyntan committed:
Added content for Level-2

kervyntan committed:
Created new class 'Task'

kervyntan committed:
1. Edited the type of the ArrayList userRequests
2. Added showing of status on list command

kervyntan committed:
Refactored if conditions into a switch case

kervyntan committed:
Added processing for userInput to handle mark/unmark case

kervyntan committed:
Added logic to handle command for mark/unmark

kervyntan committed:
Update name variable to description

kervyntan committed:
Created different types of Task - ToDo, Deadline and Event

kervyntan committed:
Created Type Enum and used it in all types of task constructors

kervyntan committed:
Huge commit - Refactored/abstracted much of the logic into respective functions, added more logic to fit for the tasks, adding appropriate getters to the different types of Tasks

kervyntan committed:
Edited runtest.sh, ran tests and updated input.txt, EXPECTED.txt based on results

kervyntan committed:
added error handling for Kervyn.java

kervyntan committed:
Added error handling for mark, unmark and delete commands, added delete command logic

kervyntan committed:
Clean up code, refactor and remove unused variable(s)

kervyntan committed:
remove variable char

kervyntan committed:
use of constant variable for chatbot name

kervyntan committed:
Amended EXPECTED test from recent commits

kervyntan committed:
Create Writer class

kervyntan committed:
Ability to write/save data to text file

kervyntan committed:
Updated reading of tasks from text file and using it as existing list

kervyntan committed:
Resolved issues with displaying and saving Event from/to timings

kervyntan committed:
Resolved issues for Deadline input

kervyntan committed:
Handle read and write for deadline and events with new DateTime formats

kervyntan committed:
Created base Skeleton and abstracted Ui and Storage

kervyntan committed:
Added ability to load tasks form storage into a TaskList class

kervyntan committed:
Operations moved to TaskList.java

kervyntan committed:
Refactored Commands, Storage, UI, Parser and added packages

kervyntan committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

kervyntan committed:
Updated build.gradle with the appropriate class name

kervyntan committed:
Created JUnit tests for Parser, Storage classes

kervyntan committed:
Packaged and built jar

kervyntan committed:
Revert "Packaged and built jar"

This reverts commit cc92bfe9a3e8e4654c385611ff77f6601fef41db.

kervyntan committed:
Packaged and built jar

kervyntan committed:
Added JavaDocs for all Tasks, common classes

kervyntan committed:
Added JavaDocs for Commands

kervyntan committed:
Amended fixes for how I am writing to a file

kervyntan committed:
Added FindCommand.java file to Commands package

kervyntan committed:
Added toString() methods for each Task type

kervyntan committed:
Updated Javadocs, amended toString in events

kervyntan committed:
Added dependencies related to JavaFX

There are several dependencies that are needed to use JavaFX that have
been added to the build.gradle file.

kervyntan committed:
Added imports to Kervyn file, edited Launcher.java

Added empty constructor, necessary imports and code as per guide for
JavaFX, and amended Launcher.java class to execute from Duke to Kervyn.

kervyntan committed:
Added basic skeleton for chatbot GUI

Utilised the tutorial for JavaFX and adapted it slightly to current
implementation

kervyntan committed:
Passed in new arguments to each command

Pass the user image for the DialogBox to each command.

kervyntan committed:
Changed argument name from userImage to kervynImage

Edited the argument name for the argument passed into commands.

kervyntan committed:
Added kervynDialog for ListCommand.java

Added the kervynDialog, fixed the path for Storage.

Apzure commented on src/main/java/kervyn/Storage.java:
```diff
@@ -0,0 +1,127 @@
+            }
+
+            FileWriter fw = new FileWriter(file.getAbsoluteFile(), false);
+//            BufferedWriter bw = new BufferedWriter(fw);
```
Comments should be indented relative to their position in the code. Also, it seems like this comment can just be removed.

Apzure commented on src/main/java/kervyn/Tasks/Task.java:
```diff
@@ -0,0 +1,98 @@
+ */
+public class Task {
+    private String description;
+    private boolean status;
```
Boolean variables/methods should be named to sound like booleans

Apzure commented on src/test/java/kervyn/ParserTest.java:
```diff
@@ -0,0 +1,41 @@
+    String[] unMarkTaskUserInput = "unmark 1".split(" ");
+    TaskList tasklist = new TaskList();
+    @Test
+    public void parseCommand_caseList() {
```
Should be camelCase

Apzure commented on src/test/java/kervyn/StorageTest.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
```
Add a line for spacing

Apzure reviewed (commented):
Looks good!
Just gave some code standard suggestions.

kervyntan committed:
Added DialogContainers for each of the commands

Added arguments for dialogContainer to all of the different commands.

kervyntan committed:
Fixed status retrieval in Storage.java, TaskList.java

Fixed status returning from readTasks method, and dialogContainers
for each method in TaskList

kervyntan committed:
Moved Ui code to Ui.java

Move code related to frame and dialog to Ui.java

lynnetteeee commented on src/main/java/kervyn/Parser.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    public String deduceCommand(String userInput, TaskList taskList, Image kervynImage, VBox dialogContainer) {
+        String[] processedUserInput = userInput.split(" ");
+        switch (processedUserInput[0]) {
```
Can fix the indentations here, case statements aren't supposed to have indentations! :)

lynnetteeee commented on src/main/java/kervyn/Tasks/TaskList.java:
```diff
@@ -0,0 +1,241 @@
+            Task task = userTasks.get(i);
+            char check = task.getStatus() ? 'X' : ' ';
+            char type = task.getCapitalType();
+            switch (type) {
```
same thing, indentations for switch and case statements!

lynnetteeee commented on src/main/java/kervyn/Tasks/TaskList.java:
```diff
@@ -0,0 +1,241 @@
+    /**
+     * Private method to handle the scenario when a task is already marked.
+     */
+    private void taskAlreadyMarked(VBox dialogContainer, Image kervynImage) {
```
minor issue, but since this method is an action/handles something (and is not a noun/thing), can consider renaming to start with a verb, e.g. `handleMarkedTask`

lynnetteeee commented on src/main/java/kervyn/Tasks/TaskList.java:
```diff
@@ -0,0 +1,241 @@
+    /**
+     * Private method to handle the scenario when a task is already unmarked.
+     */
+    private static void taskAlreadyUnMarked(VBox dialogContainer, Image kervynImage) {
```
Similarly, can consider using a verb name for the method, e.g. `handleUnmarkedTask`.

lynnetteeee commented on src/main/java/kervyn/Tasks/TaskList.java:
```diff
@@ -0,0 +1,241 @@
+        }
+    }
+
+    public void findTask(ArrayList<Task> userTasks, String userInput, Image kervynImage, VBox dialogContainer) {
```
I think its more apparent in this class, just wanted to point out your use of similar naming conventions for related methods! e.g. `markTask`, `unMarkTask`, `findTask`, `removeTask` - naming these methods similarly does help make your code readable! good job!

lynnetteeee reviewed (commented):
Generally LGTM! Just need to take note of indentations of switch and case statements, and some of the namings - usage of nouns/verbs! Other than these minor fixes, good job!

johnyoozhengxian commented on src/main/java/kervyn/Commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,110 @@
+/**
+ * Represents the "Deadline" command in the application, used to create and add a Deadline task to the TaskList.
+ */
+public class DeadlineCommand extends Command {
```
good use of oop

johnyoozhengxian commented on src/main/java/kervyn/Commands/UnMarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+/**
+ * Represents the "UnMark" command in the application, used to mark a task as not completed in the TaskList.
+ */
+public class UnMarkCommand extends Command {
```
good naming, can understand easily what the command does

johnyoozhengxian commented on src/main/java/kervyn/Ui.java:
```diff
@@ -0,0 +1,118 @@
+ * The Ui class is responsible for handling user interactions.
+ * It captures user input and displays output from the chatbot.
+ */
+public class Ui {
```
UI commands make sense

johnyoozhengxian reviewed (commented):
Very good chatbot created. Easy to understand code with good functions

kervyntan committed:
Added assertions in Parser.java, Storage.java

As a step towards writing higher quality code and a smoother debugging
process, I have added the necessary assertions in the above 2 files.

kervyntan committed:
Merge pull request #1 from kervyntan/branch-A-Assertions

Added assertions in Parser.java, Storage.java

kervyntan committed:
Changed name of boolean variable to fit coding standards

Edited name of boolean variable from "status" to "isCompleted" to fit
coding standards more accurately.

kervyntan committed:
Edited names of boolean variables within Deadline.java, Task.java

Refactored names of variables ("status") to match isCompleted variable
name.

kervyntan committed:
Merge pull request #2 from kervyntan/branch-A-CodeQuality

Branch A-CodeQuality

kervyntan committed:
Renamed boolean variable and ensured Find searches more finely

kervyntan committed:
Created basic deploy.yml file

Created basic .yml file based on SE website and GitHub Actions
documentation.

kervyntan committed:
Edited ParserTest.java, StorageTest.java

Commented out logic within test functions that are not updated to fit
latest code.

kervyntan committed:
Added Ui.png

Added snapshot of GUI to docs folder.

kervyntan committed:
Edited README.md

Added UserGuide with respective screenshots for demo purposes to README.

kervyntan committed:
Added file creation in Storage.java

Added logic in readTasks for Storage.java for jar to work

kervyntan committed:
Updated ParserTest.java, StorageTest.java and other files

Updated test cases in JUnit test files based on latest implementations.

kervyntan committed:
Updated build.gradle file

kervyntan committed:
Updated build.gradle
